client/src/components/ActiveChat/Input.js

```diff
@@ -39,11 +53,56 @@ class Input extends Component {
       recipientId: this.props.otherUser.id,
       conversationId: this.props.conversationId,
       sender: this.props.conversationId ? null : this.props.user,
+      attachments: this.state.attachments,
     };
-    if (reqBody.text !== "") await this.props.postMessage(reqBody);
-    this.setState({
-      text: "",
-    });
+    // only post message when it's not an empty string, unless there are attachments to that message
+    if (
+      reqBody.attachments.length > 0 ||
+      (reqBody.text !== "" && reqBody.attachments.length === 0)
+    ) {
+      await this.props.postMessage(reqBody);
+      // empty the state (text and attachments) every time a message is posted
+      this.setState({
+        text: "",
+        attachments: [],
+      });
+    }
+  };
+
+  uploadImage = (files) => {
+    const formData = new FormData();
+    const attachments = [];
+
+    // function that handles each promise post request from Cloudinary
+    const promiseForAttachmentUrl = (attachment) => {
+      formData.append("file", attachment);
+      formData.append("upload_preset", "vgnox97i");
+
+      return fetch("https://api.cloudinary.com/v1_1/dtcgl7plw/image/upload", {
+        method: "POST",
+        body: formData,
+      });
+    };
+
+    for (let idx in files) {
+      attachments.push(promiseForAttachmentUrl(files[idx]));
+    }
+
+    // only execute after all the images uploaded were fetched from Cloudinary
+    Promise.all(attachments)
+      .then((response) => {
+        return Promise.all(response.map((res) => res.json()));
+      })
+      .then((dataArr) => {
+        dataArr.forEach((data) => {
+          // only push the attachment to attachments array if url exists
+          data.url &&
+            this.setState((prevState) => ({
+              attachments: [data.url, ...prevState.attachments],
+            }));
+        });
+        alert("Attachments successfully uploaded! Press 'enter' to submit.");
+      });
```

Comment:
You should use the `async` `await` keywords to be consistent with the rest of the codebase. See [this document](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#awaiting_a_promise.all) for reference :)